.github/workflows/kvrocks.yaml

```diff
@@ -65,10 +64,12 @@ jobs:
         if: always() && steps.check-format.outcome != 'success'
         with:
           path: clang-format.patch
-      - name: Lint and check code
+      - name: Lint gocase code
+        run: ./x.py check golangci-lint
+      - name: Check cpp code via clang-tidy
         run: |
-          ./x.py check cppcheck
-          ./x.py check golangci-lint
+          ./x.py build tidy-build --config-only
+          ./x.py check tidy tidy-build -j $(nproc) --clang-tidy-path clang-tidy-12 --run-clang-tidy-path run-clang-tidy-12
```

Comment:
Does clang-tidy depend on a configured source? These steps seems too complex to a new contributor to understand.

Reply:
Yeah, clang-tidy need a full configuration, which means we should run cmake to generate all build informations.

I think it is not a complex proceudure, since as a developer, we always build the project by our own, e.g.
```
./x.py build
```
Then, if want to run clang-tidy, we can just run:
```
./x.py check tidy # the `build` dir can be omitted 
```

Reply:
Can clang-tidy auto fix violations?

Reply:
Just some of clang-tidy checkers provide the auto-fix functionality, some of which is not perfect auto-fix, we need to adjust the fixed code to make sure the fix is right. All `clang-analyzer-*` checkers cannot auto-fix, e.g. it can detect a null pointer dereference (NPE in java), but it can hardly know how to fix it (it is a indeed hard problem for computer :rofl:)

Reply:
And the auto-fix message will appear on the report message, so I prefer manually apply them, since the clang-tidy is bounded to most IDEs and editors, e.g. vim/emacs/vscode via clangd, clion, visual studio, we can see these error/warning on our editor, and some IDEs provide a fix button.

Reply:
So actually, when we are using professional C++ IDE (e.g. clion) or well-configured editor (e.g. vim/emacs/vscode with clangd), we do not need to run `./x.py check tidy`, and these report will always appear in the editor, just like compile warning/errors which will be integrated to the editor box.